websockets/cext.c

```diff
@@ -0,0 +1,128 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <stdlib.h> /* getenv */
+
+#ifndef HAS_SSE2
+/* setup HAS_SSE2 macro according to compiler predefined macros */
+#ifdef __SSE2__
+#define HAS_SSE2 1
+#else
+#define HAS_SSE2 0
+#endif
+#endif /* not defined HAS_SSE2 */
+
+#if HAS_SSE2
+#include <emmintrin.h>
+#endif
+
+#ifndef HAS_FAST_UNALIGNED_ACCESS
+/* This is not enabled by default */
+/* build with CFLAGS="-DHAS_FAST_UNALIGNED_ACCESS=1" to enable */
```

Comment:
Here's the benchmark. Of course, this makes sense only when SSE2 is not available (i386, or ARM, MIPS, PowerPC)
As said in the comment, there seem to be no normative information regarding alignment of PyBytes allocated buffer. Given the code of cpython, the buffer will always be aligned properly. I don't know about other implementations.

```
SSE2=0
HAS_FAST_UNALIGNED_ACCESS=0
pure_python_tobytearray:  12295.876 MB/s (6,291,456 bytes)
pure_python_old:              4.874 MB/s (6,291,456 bytes)
pure_python_new:              7.125 MB/s (6,291,456 bytes)
apply_mask:                1682.360 MB/s (6,291,456 bytes)
pure_python_tobytearray:  19080.777 MB/s (60,000 bytes)
pure_python_old:              4.834 MB/s (60,000 bytes)
pure_python_new:              7.552 MB/s (60,000 bytes)
apply_mask:                1975.460 MB/s (60,000 bytes)
pure_python_tobytearray:     16.923 MB/s (6 bytes)
pure_python_old:              2.858 MB/s (6 bytes)
pure_python_new:              2.993 MB/s (6 bytes)
apply_mask:                  24.933 MB/s (6 bytes)
pure_python_tobytearray:      2.856 MB/s (1 bytes)
pure_python_old:              0.896 MB/s (1 bytes)
pure_python_new:              0.824 MB/s (1 bytes)
apply_mask:                   4.189 MB/s (1 bytes)

SSE2=0
HAS_FAST_UNALIGNED_ACCESS=1
pure_python_tobytearray:  12382.198 MB/s (6,291,456 bytes)
pure_python_old:              5.141 MB/s (6,291,456 bytes)
pure_python_new:              7.593 MB/s (6,291,456 bytes)
apply_mask:               10109.721 MB/s (6,291,456 bytes)
pure_python_tobytearray:  21653.598 MB/s (60,000 bytes)
pure_python_old:              5.060 MB/s (60,000 bytes)
pure_python_new:              7.383 MB/s (60,000 bytes)
apply_mask:               13648.743 MB/s (60,000 bytes)
pure_python_tobytearray:     17.094 MB/s (6 bytes)
pure_python_old:              2.658 MB/s (6 bytes)
pure_python_new:              2.771 MB/s (6 bytes)
apply_mask:                  24.282 MB/s (6 bytes)
pure_python_tobytearray:      2.686 MB/s (1 bytes)
pure_python_old:              0.890 MB/s (1 bytes)
pure_python_new:              0.826 MB/s (1 bytes)
apply_mask:                   3.837 MB/s (1 bytes)
```